prometheus-ksonnet/lib/prometheus-ha-mixin.libsonnet

```diff
@@ -0,0 +1,42 @@
+{
+  prometheus_config_0: $.prometheus_config,
+  prometheus_config_1: $.prometheus_config,
+
+  prometheusAlerts_0: $.prometheusAlerts,
+  prometheusAlerts_1: $.prometheusAlerts,
+
+  prometheusRules_0: $.prometheusRules,
+  prometheusRules_1: $.prometheusRules,
+
+  prometheus+:: {
+    local configMap = $.core.v1.configMap,
+
+    prometheus_config_map:: {},
+
+    prometheus_config_maps: [{
+      configMap.new('%s-config-0' % self.name) +
+      configMap.withData({
+        'prometheus.yml': $.util.manifestYaml($.prometheus_config_0),
+        'alerts.rules': $.util.manifestYaml($.prometheusAlerts_0),
+        'recording.rules': $.util.manifestYaml($.prometheusRules_0),
+      }),
+      configMap.new('%s-config-1' % self.name) +
+      configMap.withData({
+        'prometheus.yml': $.util.manifestYaml($.prometheus_config_1),
+        'alerts.rules': $.util.manifestYaml($.prometheusAlerts_1),
+        'recording.rules': $.util.manifestYaml($.prometheusRules_1),
+      }),
+    ]},
+
+  prometheus_config_mount:: {},
+
+  prometheus_init_container::
+    container.new('prometheus-init', 'busybox') +
+    container.withCommand(["ln", "-s", "/etc/$HOSTNAME", "/etc/prometheus"]),
```

Comment:
Does this work?  I'd be surprised it the `/etc` directory is shared between the init container and the main container.

That being said, you could mount a tmp volume that is shared...

Reply:
As I said above, I havent tried this :-) that comes next, and there may well be issues like the one you mention.

Reply:
Give us a ping when you've tested it and I'll do another review.